scipy/_lib/_util.py

```diff
@@ -337,3 +338,73 @@ def getargspec_no_self(func):
         if argspec.args[0] == 'self':
             argspec.args.pop(0)
         return argspec
+
+
+class PoolWrapper(object):
+    """
+    Wrapper for working with objects that have map-like methods, such as
+    `multiprocessing.Pool`. Used for parallelisation.
+
+    Parameters
+    ----------
+    pool : int or map-like object
+        If `pool` is an `int` then it specifies the number of threads to
+        use for parallelization. If `int(pool) in [0, 1]`, then no parallel
```

Comment:
why allow `workers=0`? better to raise an error for anything that's not a positive integer or -1.

Reply:
I was thinking it was easier for a user, but perhaps it makes sense to tighten up the requirements.